### DISCUSSION TITLE
lava not burning items on death

### DISCUSSION THREAD

**Discussion by elaineandsparky** [GriefPrevention/GriefPrevention#1106] [Q&A]
I'm 99% sure we never used to have this problem but it has been brought to my attention that if you die in lava the items you drop on death do not burn and can be picked back up on doing /back. I want lava to burn items on death!
Lava does however destroy items you chuck into it, it is just on death that is the issue.

 ProtectItemsDroppedOnDeath:
    PvPWorlds: true
    NonPvPWorlds: true

changing the nonpvpworlds to false in the configs does make lava burn items on death but it also allows other players to pick up your items if you die by other means which I do not want people to be able to do.

I'm stuck for ideas on how to fix this.

**Comment by bigpresh** [comment]
It looks likely that it's being blocked by this code:
https://github.com/TechFortress/GriefPrevention/blob/97cbd80a707324a5a858b020b7d3b508d6a47ad5/src/main/java/me/ryanhamshire/GriefPrevention/EntityEventHandler.java#L721-L729

That hasn't changed for quite some time.

I think the logic is that, if you die in PvP and your drops are supposed to be protected, a vindictive player can't just drop a bucket of lava to burn them up, which seems to fit with the idea of protecting them.

I guess a new setting to tell GP to protect drops only if you died from PvP damage could be useful - but could have weird edge cases - e.g. what if you died because someone hit you with knockback and sent you off a cliff - should the drops be protected then?

This could easily be handled in an addon as GP fires a `ProtectDeathDropsEvent` which is cancellable... although it doesn't pass on the `EntityDeathEvent` event, which would have made the addon code's life a lot easier...

**Comment by elaineandsparky** [comment]
that logic makes sence for a pvp world but not a nonpvp world where dumping lava close to a player is also blocked.

Either way thank you for your reply and i'm assuming it isn't anything I can change my end? (i'm still learning configs) because at this point i may as well have keep inv on which i think my players would quite like but not what i intended for the server lol.

**Comment by RoboMWM** [comment]
pretty sure there's a config to lock in either pvp or pve
